docs/reST/ref/math.rst

```diff
@@ -406,8 +406,12 @@ Multiple coordinates can be set using slices or swizzling
       | :sg:`clamp_magnitude(max_length) -> Vector2`
       | :sg:`clamp_magnitude(min_length, max_length) -> Vector2`
 
-      Returns a new copy of a vector with the magnitude clamped between max_length and min_length. If
-      a min_length value is not assigned, then it will automatically be set to 0.
+      Returns a new copy of a vector with the magnitude clamped between 
+      ``max_length`` and ``min_length``. If only one argument is passed, it is 
+      taken to be the ``max_length``
+
+      This function raises ``ValueError`` if ``min_length`` is greater than
+      ``max_length``, or if either of these values are negative.
 
```

Comment:
I think we should mark new APIs Experimental in general. Then after some releases(and after more feedback) the experimental marking can be removed from the docs. What do you think?

Reply:
While it does sound nice to mark them as experimental in the docs (and we should do it), I don't think a lot of people would just find it in there. They might see it in a forum or something similar, go "oh cool, I could use that", and then the next time they upgrade, their code is going to break.

Just putting out ideas here, but maybe we could do a system similar to pythons future module, where one could enable specific features one by one? So something like this: 
```py
import pygame
from pygame.future import vector_clamp  # or some other module name

vector_clamp.enable()  # commenting this line will cause the code below to raise AttributeError

vec = pygame.Vector2(10, 20)
vec.clamp_magnitude(5)
```

However this might be too much effort to implement, might not be elegant (in pygames code base) and could also make new contributors a bit confused

Reply:
Good points/ideas.

Another problem with experimental stuff is people abandon it and don't finish the work. Leaving their mess to other people to clean up. I've a feeling this would help some people push things over the line to be finished.

Probably the feature flag/future idea should move to a separate issue.

Reply:
Done, these functions are now marked experimental in the docs